Sources/FileSystem/FileSystemTestingTrait.swift

```diff
@@ -0,0 +1,32 @@
+#if DEBUG
+    #if canImport(Testing)
+        import Path
+        import Testing
+
+        extension FileSystem {
+            /// It returns the temporary directory created when using the `@Test(.inTemporaryDirectory)`.
+            /// Note that since the value is only propagated through Swift structured concurrency, if you use DispatchQueue,
+            /// values won't be propagated so you'll have to make sure they are explicitly passed down.
+            @TaskLocal public static var testTemporaryDirectory: AbsolutePath?
```

Comment:
```suggestion
            @TaskLocal public static var temporaryTestDirectory: AbsolutePath?
```

nit: I believe this reads slightly better.